### PR TITLE
feat(tasks):end-to-end tests for the task system

### LINKED ISSUE #11360: Add task test to influxdb/cmd/influxd (Emrys)
Add end to end tests for tasks in influxdb.

### PR DESCRIPTION
**Pull request by docmerlin**
Closes #11360

_Briefly describe your proposed changes:_

Adds some end to end tests for tasks using the launcher.

  - [x ] Rebased/mergeable
  - [x ] Tests pass

### CODE REVIEW

**Comment by mark-rushakoff** [comment]
I've been asked that we wait to merge this until after the alpha cut, so I'll review it after that.

File: cmd/influxd/launcher/tasks_test.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+	"github.com/google/go-cmp/cmp"
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/execute/executetest"
+	platform "github.com/influxdata/influxdb"
+	pctx "github.com/influxdata/influxdb/context"
+	"github.com/influxdata/influxdb/task/backend"
+)
+
+func TestLauncher_Task(t *testing.T) {
+	//ctx := context.Background()
```
Comment (mark-rushakoff) [inline cmd/influxd/launcher/tasks_test.go]: Remove comment.

File: cmd/influxd/launcher/tasks_test.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+	}
+
+	ctx = pctx.SetAuthorizer(context.Background(), be.Auth)
+
+	be.Auth = &platform.Authorization{UserID: u.ID, OrgID: org.ID, Permissions: []platform.Permission{*writeBIn, *writeBOut, *writeT, *readT}}
+	if err := be.AuthorizationService().CreateAuthorization(context.Background(), be.Auth); err != nil {
+		t.Fatal(err)
+	}
+	if !be.Org.ID.Valid() {
+		panic("invalid org id")
```
Comment (mark-rushakoff) [inline cmd/influxd/launcher/tasks_test.go]: This can be t.Fatal.

File: cmd/influxd/launcher/tasks_test.go [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+	ndrString := time.Unix(ndr, 0).UTC().Format(time.RFC3339)
+	var targetRun platform.Run
+	i := 0
+	for {
+		t.Logf("Looking for created run...")
+		if time.Now().After(deadline) {
+			t.Fatalf("didn't find completed run within deadline")
+		}
+		time.Sleep(5 * time.Millisecond)
+		//pipeline.Flush() // To ensure we can see writes from the task run logs.
```
Comment (mark-rushakoff) [inline cmd/influxd/launcher/tasks_test.go]: Uncomment or remove?

File: cmd/influxd/launcher/tasks_test.go [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+	for _, g := range got {
+		executetest.NormalizeTables(g)
+	}
+
+	if !cmp.Equal(got, want) {
+		t.Fatal("unexpected task result -want/+got", cmp.Diff(want, got))
+	}
+
+	t.Run("showrun", func(t *testing.T) {
+		t.Skip("FindRunByID isn't returning the log")
```
Comment (mark-rushakoff) [inline cmd/influxd/launcher/tasks_test.go]: Should this still be skipped?

File: cmd/influxd/launcher/tasks_test.go [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+		// do a show run!
+		showRun, err := be.TaskService().FindRunByID(ctx, created.ID, targetRun.ID)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !cmp.Equal(targetRun, *showRun) {
+			t.Fatal("unexpected task result -want/+got\n", cmp.Diff(targetRun, showRun))
+		}
+
```
Comment (mark-rushakoff) [inline cmd/influxd/launcher/tasks_test.go]: Remove blank line.

File: task/platform_adapter.go [hunk 6]
```diff
@@ -17,19 +17,19 @@ type RunController interface {
...
 func PlatformAdapter(s backend.Store, r backend.LogReader, rc RunController) platform.TaskService {
-	return pAdapter{s: s, r: r, rc: rc}
+	return pAdapter{S: s, R: r, RC: rc}
 }
 
 type pAdapter struct {
-	s  backend.Store
-	rc RunController
-	r  backend.LogReader
+	S  backend.Store
```
Comment (mark-rushakoff) [inline task/platform_adapter.go]: These fields shouldn't be exported. If you need to hold a reference to them, keep the reference from before the call to `PlatformAdapter`.

File: cmd/influxd/launcher/tasks_test.go (hunk 4 above)
Comment (docmerlin) [reply cmd/influxd/launcher/tasks_test.go]: Yes, it is still not fixed.

File: cmd/influxd/launcher/query_test.go [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+		total++
+		b.Do(func(c flux.ColReader) error { return nil })
+		return nil
+	})
+	return total
+}
+
+// MustExecuteQuery executes the provided query panicking if an error is encountered.
+// Callers of MustExecuteQuery must call Done on the returned QueryResults.
+func (p *Launcher) MustExecuteQuery(orgID platform.ID, query string, auth *platform.Authorization) *QueryResults {
```
Comment (mark-rushakoff) [inline cmd/influxd/launcher/query_test.go]: Just a note, adding methods to a type from a test package can cause some confusing errors later. The alternative would be to move the receiver into the argument list. But I can live with it for now.

Comment (docmerlin) [reply cmd/influxd/launcher/query_test.go]: This type is different from the Launcher in launcher.go, it is a test wrapper around the other Launcher.

**Pull request merged**